src/main/java/ai/elimu/web/SignOnControllerDiscord.java

```diff
@@ -0,0 +1,188 @@
+package ai.elimu.web;
+
+import java.io.IOException;
+import java.util.Calendar;
+import javax.servlet.http.HttpServletRequest;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONObject;
+import ai.elimu.dao.ContributorDao;
+import ai.elimu.model.contributor.Contributor;
+import ai.elimu.model.v2.enums.Environment;
+import ai.elimu.model.enums.Role;
+import ai.elimu.util.ConfigHelper;
+import ai.elimu.web.context.EnvironmentContextLoaderListener;
+import com.github.scribejava.apis.DiscordApi;
+import com.github.scribejava.core.builder.ServiceBuilder;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.OAuthRequest;
+import com.github.scribejava.core.model.Response;
+import com.github.scribejava.core.model.Verb;
+import com.github.scribejava.core.oauth.OAuth20Service;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+
+import org.springframework.web.bind.annotation.RequestMethod;
+
+
+@Controller
+public class SignOnControllerDiscord {
+
+    private static final String NETWORK_NAME = "discord";
+
+    private static final String PROTECTED_RESOURCE_URL = "https://discordapp.com/api/users/@me";
+
+    private OAuth20Service oAuth20Service;
+
+    private final String secretState = "secret_" + new Random().nextInt(999_999);
+    
+    private Logger logger = LogManager.getLogger();
+    
+    @Autowired
+    private ContributorDao contributorDao;
+
+    @RequestMapping("/sign-on/discord")
+    public String handleAuthorization(HttpServletRequest request) throws IOException {
+        logger.info("handleAuthorization");
+        String clientId = "978873481649348668";
+        String clientSecret = "Lf0O65Zk0Qv9cXyC7igPgljIjSJJBZbv";
```

Comment:
Should the clientId and clientSecret moved to config/env file instead of keeping it in code?

Reply:
> Should the clientId and clientSecret moved to config/env file instead of keeping it in code?

@jpatel3 Yes. This code is for the development environment, so we included the secret there (so that developers won't have to spend time setting it up themselves).

Made a small refactoring to make this part of the code a little bit more easy to read: https://github.com/elimu-ai/webapp/commit/c5175e22374eac938e10c1176a8ac47110d1f77a